pkgs/build-support/binary-cache.nix

```diff
@@ -0,0 +1,77 @@
+{ stdenv, buildPackages }:
+
+# This function is for creating a flat-file binary cache, i.e. the kind created by
+# nix copy --to file:///some/path and usable as a substituter (with the file:// prefix).
+
+# For example, in the Nixpkgs repo:
+# nix-build -E 'with import ./. {}; mkBinaryCache { rootPaths = [hello]; }'
+
+{ name ? "binary-cache"
+, rootPaths
+}:
+
+stdenv.mkDerivation {
+  inherit name;
+
+  __structuredAttrs = true;
```

Comment:
Was `__structuredAttrs` ever stabilized?  I can't find it in the manual and it only seems to be used in two places in nixpkgs, both of which are `build-support` routines that aren't used by anything else (i.e. they are unreachable).

There are a couple of other comments elsewhere in nixpkgs about `__structuredAttrs` not being available.

Reply:
I have no idea, this is mimicking `closureInfo`

Reply:
It's ok to use it here.

https://github.com/NixOS/nixpkgs/pull/175649 is the current effort to move more of Nixpkgs towards structured attrs.